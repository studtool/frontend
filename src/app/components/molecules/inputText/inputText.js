import React, {Component} from 'react';
import {Label} from '../../atoms/label/label';
import {InputTextField} from '../../atoms/inputTextField/inputTextField.js';

export class InputText extends Component {

    render() {
        const {
            children,
            ...other
        } = this.props;

        const qa = this.props.qa ? this.props.qa : 'no-qa';
        return (
            <>
                <div qa={qa}>
                    <Label>
                        {children}

                        <InputTextField
                            {...other}
                        />
                    </Label>
                </div>
            </>
        );
    }
}
