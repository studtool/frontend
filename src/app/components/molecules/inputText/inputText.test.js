import React from 'react';
import {InputText} from './inputText';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

test('inputText test', () => {
    const stubFunc = () => {};

    const tree = renderer.create(
        <InputText
            qa='test-input'
            name="testName"
            type="password"
            value="thisValue"
            onChange={stubFunc}
        >
            test
        </InputText>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
