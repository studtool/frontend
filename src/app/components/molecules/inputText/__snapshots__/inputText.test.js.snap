// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputText test 1`] = `
<div
  qa="test-input"
>
  <label
    className=""
  >
    test
    <input
      className=""
      name="testName"
      onChange={[Function]}
      type="password"
      value="thisValue"
    />
  </label>
</div>
`;
